toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -372,6 +521,30 @@ func parseSourcePartition(source string) (imagecustomizerapi.MountIdentifierType
 	return imagecustomizerapi.MountIdentifierTypeDefault, "", err
 }
 
+func parseExtendedSourcePartition(source string) (imagecustomizerapi.ExtendedMountIdentifierType, string, error) {
```

Comment:
can we update `parseSourcePartition` instead of creating a new function?
are there scenarios where the new `/dev` should not be handled and an error must be still returned?